### PR TITLE
Add PyTorch Lightning integration

### LINKED ISSUE #195: `torch` / `lightning` integration
It would be nice to have integration with basic `torch` and `lightning` tuning workflows, to allow autoML style tuning.

This would require a `BaseExperiment` descendant class `TorchExperiment` which takes a `DataLoader`, a `LightningModule`, and a `Trainer`, performs a training run, and returns the validation loss/score.

Since tuning is not API preserving in `torch` / `lightning`, I would suggest an additional function that produces parameters for a tuned network, or the initialized tuned network with said parameters.

Comments appreciated on the design.

### PR DESCRIPTION
**Pull request by amitsubhashchejara**
This pull request is to merge a new experiment `torch-lightning-experiment` for the issue `195`. The new class `TorchExperiment` is an experiment adapter for PyTorch Lightning experiments and is used to perform experiments using PyTorch Lightning modules. It allows for hyperparameter tuning and evaluation of the model's performance using specified metrics. 

The `TorchExperiment` class accepts a `LightningModule`, `DataModule`, `Trainer`, and an `objective_matric` with default value `val_loss`. The `_evaluate` function internally performs a training run and returns a score of the `objective_matric`.

Fixes #195

### CODE REVIEW

**Comment by SimonBlanke** [comment]
Great work! :-)

**Comment by amitsubhashchejara** [comment]
@SimonBlanke @fkiraly 
Thank you so much for the assistance. Let's discuss further possibilities to make the integration more robust.
